### PR TITLE
Xcode 16 & Swift 6

### PR DESCRIPTION
**Pull request by guykogus**
🚀

## Checklist
- [x] I checked the [**Contributing Guidelines**](https://github.com/SwifterSwift/SwifterSwift/blob/master/CONTRIBUTING.md) before creating this request.
- [x] New extensions are written in Swift 6.0.
- [x] New extensions support iOS 12.0+ / tvOS 12.0+ / macOS 10.13+ / watchOS 4.0+, or use `@available` if not.
- [x] I have added tests for new extensions, and they passed.
- [x] All extensions have a **clear** comments explaining their functionality, all parameters and return type in English.
- [x] All extensions are declared as **public**.
- [x] I have added a [changelog](https://github.com/SwifterSwift/SwifterSwift/blob/master/CHANGELOG_GUIDELINES.md) entry describing my changes.

### CODE REVIEW

File: Sources/SwifterSwift/Dispatch/DispatchQueueExtensions.swift [hunk 1]
```diff
@@ -45,16 +45,17 @@ public extension DispatchQueue {
     func asyncAfter(delay: Double,
                     qos: DispatchQoS = .unspecified,
                     flags: DispatchWorkItemFlags = [],
-                    execute work: @escaping () -> Void) {
+                    execute work: @Sendable @escaping () -> Void) {
         asyncAfter(deadline: .now() + delay, qos: qos, flags: flags, execute: work)
     }
 
+    @available(macOS 14.0, iOS 17.0, tvOS 17.0, watchOS 10.0, *)
```
Comment (guykogus) [inline Sources/SwifterSwift/Dispatch/DispatchQueueExtensions.swift:52]: Maybe this can be done better...

**Review by LucianoPAlmeida — APPROVED** [review]

**Comment by guykogus** [comment]
@LucianoPAlmeida I think maybe we'll leave this PR open until Xcode 16 is officially released. WDYT? Otherwise it forces anyone on the `master` branch to upgrade to Swift 6, which is not a fun task if you've tried it on your own projects...

**Comment by LucianoPAlmeida** [comment]
> @LucianoPAlmeida I think maybe we'll leave this PR open until Xcode 16 is officially released. WDYT? Otherwise it forces anyone on the `master` branch to upgrade to Swift 6, which is not a fun task if you've tried it on your own projects...

Let's wait then, at least for the RC :)

**Comment by guykogus** [comment]
@LucianoPAlmeida OK, looks like _finally_ ready to merge. After this shall we release version 7.0.0?

**Review by LucianoPAlmeida — APPROVED** [review]
Awesome!
Let's do it :)

**Pull request merged**